openmdao/core/tests/test_distrib_derivs.py

```diff
@@ -629,6 +629,60 @@ def test_distribcomp_zerolengthinputsoutputs(self):
         assert_near_equal(J[('C2.outvec', 'indep.x')], 
                           np.eye(3)*np.append(2*np.ones(1), -3*np.ones(2)))
 
+class DistribCompDenseJac(om.ExplicitComponent):
+    def initialize(self):
+        self.options['distributed'] = True
+        self.options.declare('size', default=7)
+
+    def setup(self):
+        N = self.options['size']
+        rank = self.comm.rank
+        self.add_input('x', shape=1, src_indices=rank)
+        sizes, offsets = evenly_distrib_idxs(self.comm.size, N)
+        self.add_output('y', shape=sizes[rank])
+        # automatically infer dimensions without specifying rows, cols
+        self.declare_partials('y', 'x')
+
+
+    def compute(self, inputs, outputs):
+        N = self.options['size']
+        rank = self.comm.rank
+        sizes, offsets = evenly_distrib_idxs(self.comm.size, N)
+        outputs['y'] = -2.33 * inputs['x'] * np.ones((sizes[rank],))
+
+    def compute_partials(self, inputs, J):
+        N = self.options['size']
+        rank = self.comm.rank
+        sizes, offsets = evenly_distrib_idxs(self.comm.size, N)
+        # Define jacobian element by element with variable size array
+        J['y','x'] = -2.33 * np.ones((sizes[rank],))
+
+class DeclarePartialsWithoutRowCol(unittest.TestCase):
+    N_PROCS = 3
+    size = 7
```

Comment:
Everything from here (size =7) onward should go inside of a method that starts with "test_".